alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -86,28 +91,39 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
         val index = docLevelMonitorInput.indices[0]
         val queries: List<DocLevelQuery> = docLevelMonitorInput.queries
 
+        val monitorMetadata = if (monitor.metadataId.isNullOrEmpty()) createMonitorMetadata(monitor.id)
+        else {
+            try {
+                getMonitorMetadata(monitorCtx.client!!, monitorCtx.xContentRegistry!!, monitor.metadataId)
+            } catch (e: Exception) {
+                logger.warn("Monitor metadata with id, ${monitor.metadataId}, is missing for monitor: ${monitor.id}")
+                createMonitorMetadata(monitor.id)
+            }
+        }
+
         val isTempMonitor = dryrun || monitor.id == Monitor.NO_ID
-        val lastRunContext = if (monitor.lastRunContext.isNullOrEmpty()) mutableMapOf()
-        else monitor.lastRunContext.toMutableMap() as MutableMap<String, MutableMap<String, Any>>
+        val lastRunContext = if (monitorMetadata.lastRunContext.isNullOrEmpty()) mutableMapOf()
+        else monitorMetadata.lastRunContext.toMutableMap() as MutableMap<String, MutableMap<String, Any>>
+
         val updatedLastRunContext = lastRunContext.toMutableMap()
 
         val queryToDocIds = mutableMapOf<DocLevelQuery, MutableSet<String>>()
+        val inputRunResults = mutableMapOf<String, MutableSet<String>>()
         val docsToQueries = mutableMapOf<String, MutableList<String>>()
-        val idQueryMap = mutableMapOf<String, DocLevelQuery>()
 
         try {
-            val getAliasesRequest = GetAliasesRequest(index)
-            val getAliasesResponse: GetAliasesResponse = monitorCtx.client!!.suspendUntil {
-                monitorCtx.client!!.admin().indices().getAliases(getAliasesRequest, it)
+            val getIndexRequest = GetIndexRequest().indices(index)
+            val getIndexResponse: GetIndexResponse = monitorCtx.client!!.suspendUntil {
+                monitorCtx.client!!.admin().indices().getIndex(getIndexRequest, it)
             }
-            val aliasIndices = getAliasesResponse.aliases.keys().map { it.value }
+            val indices = getIndexResponse.indices()
 
-            val isAlias = aliasIndices.isNotEmpty()
-            logger.debug("index, $index, is an alias index: $isAlias")
-
-            // If the input index is an alias, creating a list of all indices associated with that alias;
-            // else creating a list containing the single index input
-            val indices = if (isAlias) getAliasesResponse.aliases.keys().map { it.value } else listOf(index)
+            // cleanup old indices that are not monitored anymore from the same monitor
+            for (ind in updatedLastRunContext.keys) {
+                if (!indices.contains(ind)) {
+                    updatedLastRunContext.remove(ind)
+                }
+            }
```

Comment:
If the same index gets added again, the next monitor execution will loose its previous context. Tradeoff of not cleaning up is then there is no good way to stop it from growing otherwise. I am fine with setting up this behavior, however calling out here to ensure that we have thought about it.

Reply:
Yea I understand your concern. I think it is more important to clean it up as this can continuously grow without stopping. Also if an index is removed from this and readded, I think it is fair to assume that is now considered as new data coming into the monitor.

Reply:
I also created an issue to find out how to cleanup old rolled over indices with issue #444